jwk/aead.go

```diff
@@ -92,10 +102,58 @@ func (c *AEAD) decrypt(ciphertext string, key []byte) ([]byte, error) {
 		return nil, errorsx.WithStack(err)
 	}
 
-	plaintext, err := cryptopasta.Decrypt(raw, aeadKey(key))
+	plaintext, err := aesGCMDecrypt(raw, aeadKey(key), additionalData)
 	if err != nil {
 		return nil, errorsx.WithStack(err)
 	}
 
 	return plaintext, nil
 }
+
+// aesGCMEncrypt encrypts data using 256-bit AES-GCM.  This both hides the content of
+// the data and provides a check that it hasn't been altered. Output takes the
+// form nonce|ciphertext|tag where '|' indicates concatenation.
+func aesGCMEncrypt(plaintext []byte, key *[32]byte, additionalData []byte) (ciphertext []byte, err error) {
```

Comment:
What's the reason for replacing cryptopasta dependency?

Reply:
handling of additional data.